cw-orch-daemon/src/senders/cosmos.rs

```diff
@@ -29,297 +30,121 @@ use cosmrs::{
-
-        let tx_builder = TxBuilder::new(tx_body);
-
-        // We retry broadcasting the tx, with the following strategies
-        // 1. In case there is an `incorrect account sequence` error, we can retry as much as possible (doesn't cost anything to the user)
-        // 2. In case there is an insufficient_fee error, we retry once (costs fee to the user everytime we submit this kind of tx)
-        // 3. In case there is an other error, we fail
-        let tx_response = TxBroadcaster::default()
-            .add_strategy(insufficient_fee_strategy())
-            .add_strategy(account_sequence_strategy())
-            .broadcast(tx_builder, self)
-            .await?;
-
-        let resp = Node::new_async(self.channel())
-            ._find_tx(tx_response.txhash)
-            .await?;
-
-        assert_broadcast_code_cosm_response(resp)
-    }
-
-    fn address(&self) -> Result<Addr, DaemonError> {
-        Ok(Addr::unchecked(self.pub_addr_str()?))
-    }
-
-    fn msg_sender(&self) -> Result<AccountId, DaemonError> {
-        if let Some(sender) = &self.options.authz_granter {
-            Ok(sender.parse()?)
-        } else {
-            self.pub_addr()
-        }
-    }
-
-    fn chain_info(&self) -> &ChainInfoOwned {
-        &self.chain_info
-    }
-
-    fn grpc_channel(&self) -> Channel {
-        self.channel()
-    }
-
-    fn set_options(&mut self, options: Self::SenderOptions) {
-        if let Some(mnemonic) = options.mnemonic.clone() {
-            let new_sender = Sender::from_mnemonic_with_options(
-                self.chain_info.clone(),
-                self.channel(),
-                &mnemonic,
-                options,
-            )
-            .unwrap();
-            *self = new_sender;
-        } else if options.hd_index.is_some() {
-            // Need to generate new sender as hd_index impacts private key
-            let new_sender = Sender::from_raw_key_with_options(
-                self.chain_info.clone(),
-                self.channel(),
-                &self.private_key.raw_key(),
-                options,
-            )
-            .unwrap();
-            *self = new_sender
-        } else {
-            self.options = options
-        }
-    }
-
-    fn build(
-        chain_info: ChainInfoOwned,
-        grpc_channel: Channel,
-        sender_options: Self::SenderOptions,
-    ) -> Result<Self, Self::Error> {
-        if let Some(mnemonic) = &sender_options.mnemonic {
-            Sender::from_mnemonic_with_options(
-                chain_info,
-                grpc_channel,
-                &mnemonic.clone(),
-                sender_options,
-            )
-        } else {
-            Sender::new_with_options(chain_info, grpc_channel, sender_options)
-        }
-    }
+    pub chain_info: Arc<ChainInfoOwned>,
+    pub(crate) options: CosmosOptions,
+    pub secp: Secp256k1<C>,
+    // Private field to ensure the struct is not constructible outside of this module
+    _private: (),
```

Comment:
```suggestion
    // Private field to ensure the struct is not constructible outside of this module
    _private: (),
```
options already private to this module